src/nvim/globals.h

```diff
@@ -92,6 +92,10 @@
 # define SYNTAX_FNAME   "$VIMRUNTIME" _PATHSEPSTR "syntax" _PATHSEPSTR "%s.vim"
 #endif
 
+#ifndef SYNTAX_FILE
+# define SYNTAX_FILE   "$VIMRUNTIME" _PATHSEPSTR "syntax" _PATHSEPSTR "syntax.vim"
+#endif
```

Comment:
Since this is really the same as SYNTAX_FNAME, https://github.com/neovim/neovim/pull/3767 attempts to avoid the extra global.